src/Adapters/Email/SimpleMailgunAdapter.js

```diff
@@ -0,0 +1,44 @@
+import Mailgun from 'mailgun-js';
+
+let SimpleMailgunAdapter = mailgunOptions => {
+	if (!mailgunOptions || !mailgunOptions.apiKey || !mailgunOptions.domain) {
+		throw 'SimpleMailgunAdapter requires an API Key and domain.';
+	}
+	let mailgun = Mailgun(mailgunOptions);
+
+	let sendMail = ({to, subject, text}) => {
+		let data = {
+			from: mailgunOptions.fromAddress,
+			to: to,
+			subject: subject,
+			text: text,
+		}
+
+		return new Promise((resolve, reject) => {
+			mailgun.messages().send(data, (err, body) => {
+				if (typeof err !== 'undefined') {
+					reject(err);
+				}
+				resolve(body);
+			});
+		});
+	}
+
+	return Object.freeze({
+		sendVerificationEmail: ({ link, user, appName, }) => {
+			let verifyMessage =
+	      "Hi,\n\n" +
+	      "You are being asked to confirm the e-mail address " + user.email + " with " + appName + "\n\n" +
+	      "" +
+	      "Click here to confirm it:\n" + link;
+			return sendMail({ 
+        to:user.email, 
```

Comment:
My standing is that Parse.Object is a more consistent, more well known, and generally better interface for interacting with data, and that if we want to prevent mutations and/or queries in a particular adapter, we should make an immutable version of Parse.Object instead of changing the entire interface.

For the push adapter I would rather have it accept Parse.Installation but I didn't catch that change before it was merged. We can still consistently use Parse.Objects for future adapters if we decide that is the way we want to go.